plugin/PDFView.qml

```diff
@@ -132,6 +136,35 @@ SilicaFlickable {
         linkColor: Theme.highlightColor;
         pagePlaceholderColor: Theme.highlightColor;
 
+        onPageLayoutChanged: if (rememberPositionConfig.value) restorePageSettings()
```

Comment:
Sorry, maybe I was not clear enough in my explanations.

To restore the page settings, and thus use the goToPage() function, the PDFCanvas object must know each page.rect. Each page.rect are calculated on the C++ side in a separate thread. This calculation is done each time the width of the PDFCanvas object is changed, thus it is done when the PDFCanvas is first set as a child in the QML page hierarchy. If one calls goToPage() before the page.rect has been calculated (for instance when reacting at the signal documentLoaded before the thread finished is work), nothing happens.

My reasoning was the following :
* I need to call goToPage(), thus I need to know when the thread finished to calculate all the page.rect.
* looking in the C++ side, I see that pageLayoutChanged signal is indeed already emitted when the thread finishes its calculation.
* thus I connect to this thread but react only once.

The path is the following :
documentLoaded -> […] -> run thread for page layout -> emit pageLayoutChanged -> adjust page position if any storage exists for this document -> don't react anymore on later pageLayoutChanged signal.